### PR TITLE
Added valgrind test case with various self tests [v2]

### PR DESCRIPTION
**Pull request by harish-24**
Valgrind is an instrumentation framework for building dynamic analysis tools. Valgrind tools that can
automatically detect many memory management and threading bugs

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

File: toolchain/valgrind.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    The Test fetches the recent valgrind tar and runs the tests in it
+    '''
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.failures = []
+
+        for package in ['gcc', 'make']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.skip(package + ' is needed for the test to be run')
```
Comment (balamuruhans) [inline toolchain/valgrind.py]: As suggested by @PraveenPenguin you can avoid using + for string concat

Comment (harish-24) [inline toolchain/valgrind.py]: sure, did not notice this.

**Comment by harish-24** [comment]
@PraveenPenguin Pushed with changes

**Comment by harish-24** [comment]
[valgrind_job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/810341/valgrind_job.txt)

File: toolchain/valgrind.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    def setUp(self):
+        smm = SoftwareManager()
+        self.failures = []
+
+        for package in ['gcc', 'make']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.skip('%s is needed for the test to be run' % package)
+        tarball = self.fetch_asset(
+            "http://valgrind.org/downloads/valgrind-3.12.0.tar.bz2",
+            expire='7d')
```
Comment (apahim) [inline toolchain/valgrind.py]: Why are you expiring a specific version? It's not supposed to change, right?

Comment (PraveenPenguin) [inline toolchain/valgrind.py]: @apahim oh thanks i miss yes this  not going update  @harish-24 can you please have a look ?

Comment (harish-24) [inline toolchain/valgrind.py]: @apahim @PraveenPenguin  sure, removed `expire`

**Comment by PraveenPenguin** [comment]
@narasimhan-v @balamuruhans @apahim thanks for review

**Pull request merged**